pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -516,16 +516,35 @@ func validateAzure(h *validatorHandler, m *Machine) (bool, utilerrors.Aggregate)
 		errs = append(errs, field.Required(field.NewPath("providerSpec", "networkResourceGroup"), "must provide a network resource group when a virtual network or subnet is specified"))
 	}
 
+	if providerSpec.Image.ResourceID == "" &&
```

Comment:
How about we break this into a seperate function so that we can return early, similar to what was done for the GCP provider? 

It would be good to make the resourceID mutually exclusive from the rest of the fields, ie, if resource ID is set and any other field is, this should be an error, I don't think that happens currently